orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py

```diff
@@ -250,25 +254,62 @@ def _get_exported_model(self, *inputs, **kwargs):
         # correct training flag to reflect the expected behavior.
         # For example, the Dropout node in a model is dropped under eval mode.
         assert self._export_mode is not None, "Please use a concrete instance of ExecutionManager"
-
+        exported_model = None
         try:
-            with torch.no_grad(), _logger.suppress_os_stream_output(log_level=self._loglevel):
-                torch.onnx.export(self._flattened_module,
-                                  sample_inputs_as_tuple,
-                                  f,
-                                  input_names=self._input_info.names,
-                                  output_names=output_names,
-                                  opset_version=ONNX_OPSET_VERSION,
-                                  do_constant_folding=False,
-                                  training=self._export_mode,
-                                  dynamic_axes=self._input_info.dynamic_axes,
-                                  verbose=self._loglevel < _logger.LogLevel.WARNING,
-                                  export_params=False,
-                                  keep_initializers_as_inputs=True)
+            if self._onnx_export_type == torch.onnx.OperatorExportTypes.ONNX_FALLTHROUGH:
```

Comment:
instead of having if-else for choosing how to exporter, we could have a factory that builds the proper exporter based on _onnx_export_type

if-else branches couples the code in a hard way to maintain over time as features start to stack

Reply:
[peng]

Reply:
We can simplify the logic to reuse as much as previous logic. 

We will have a flag for control whether we need a deepcopy before model export. And whether we should do a post process for the exported model. 

Honestly I don't think there is a strong need introducing a factory here.